src/StaticLS/IDE/CodeActions/InsertFields.hs

```diff
@@ -0,0 +1,33 @@
+module StaticLS.IDE.CodeActions.InsertFields where
+
+import Data.Text qualified as T
+import Language.LSP.Protocol.Types qualified as LSP
+import StaticLS.IDE.CodeActions.Utils (indentation, prefer, quickFix)
+
+codeAction ::
+  LSP.TextDocumentIdentifier ->
+  LSP.Diagnostic ->
+  T.Text ->
+  Maybe T.Text ->
+  [T.Text] ->
+  LSP.CodeAction
+codeAction = insertFields
+
+insertFields ::
+  LSP.TextDocumentIdentifier ->
+  LSP.Diagnostic ->
+  T.Text ->
+  Maybe T.Text ->
+  [T.Text] ->
+  LSP.CodeAction
+insertFields tdi diag ctor existingFields missingFields =
```

Comment:
If a GHC error (or warning) tells you which fields are missing, this code action will create a quick fix that will insert the missing fields for you.